cirq/ops/phased_iswap_gate.py

```diff
@@ -0,0 +1,203 @@
+        oz = args.subspace_index(0b10)
+        args.target_tensor[zo] *= phase
+        args.target_tensor[oz] *= phase.conjugate()
+        args.target_tensor = self._iswap._apply_unitary_(args)
+        assert args.target_tensor is not None
+        args.target_tensor[zo] *= phase.conjugate()
+        args.target_tensor[oz] *= phase
+        return args.target_tensor
+
+    def _decompose_(self, qubits: Sequence[raw_types.Qid]) -> op_tree.OP_TREE:
+        if len(qubits) != 2:
+            raise ValueError(f'Expected two qubits, got {len(qubits)}')
+        a, b = qubits
+
+        yield cirq.Z(a)**self.phase_exponent
+        yield cirq.Z(b)**-self.phase_exponent
+        yield from protocols.decompose_once_with_qubits(self._iswap, qubits)
+        yield cirq.Z(a)**-self.phase_exponent
+        yield cirq.Z(b)**self.phase_exponent
+
+    def _pauli_expansion_(self) -> value.LinearDict[str]:
+        if self._is_parameterized_():
+            return NotImplemented
+        expansion = protocols.pauli_expansion(self._iswap)
+        assert set(expansion.keys()).issubset({'II', 'XX', 'YY', 'ZZ'})
+        assert np.isclose(expansion['XX'], expansion['YY'])
+
+        v = (expansion['XX'] + expansion['YY']) / 2
+        phase_angle = np.pi * self.phase_exponent
+        c, s = np.cos(2 * phase_angle), np.sin(2 * phase_angle)
+
+        return value.LinearDict({
+            'II': expansion['II'],
+            'XX': c * v,
+            'YY': c * v,
+            'XY': s * v,
+            'YX': -s * v,
+            'ZZ': expansion['ZZ'],
+        })
+
+    def __str__(self) -> str:
+        if self.exponent == 1:
+            return 'PhasedISWAP'
+        return f'PhasedISWAP**{self.exponent}'
+
+    def __repr__(self):
+        phase_exponent = proper_repr(self._phase_exponent)
+        args = [f'phase_exponent={phase_exponent}']
+        if self.exponent != 1:
+            exponent = proper_repr(self.exponent)
+            args.append(f'exponent={exponent}')
+        arg_string = ', '.join(args)
+        return f'cirq.PhasedISwapPowGate({arg_string})'
+
+
+# In numerical linear algebra Givens rotation is any linear transformation
+# with matrix equal to the identity except for a 2x2 orthogonal submatrix
+# [[cos(a), -sin(a)], [sin(a), cos(a)]] which performs a 2D rotation on a
+# subspace spanned by two basis vectors. In quantum computational chemistry
+# the term is used to refer to the two-qubit gate defined as
+#
+#     GivensRotation(a) ≡ exp(-i a (Y⊗X - X⊗Y) / 2)
+#
+# with the matrix
+#
+#     [[1, 0, 0, 0],
+#      [0, c, -s, 0],
+#      [0, s, c, 0],
+#      [0, 0, 0, 1]]
+#
+# where
+#
+#     c = cos(a),
+#     s = sin(a).
+#
+# The matrix is a Givens rotation in the numerical linear algebra sense
+# acting on the subspace spanned by the |01⟩ and |10⟩ states.
+#
+# The gate is also equivalent to the ISWAP conjugated by T^-1 ⊗ T.
+#
+def GivensRotation(angle_rads: value.TParamVal) -> PhasedISwapPowGate:
```

Comment:
Conflicted about the casing here. Per PEP should be `givens_rotation`... Hmm.

Reply:
We kind of violated the PEP with gate factories such as `Rz`. I think we did that to make those not stand out among other gates, e.g. in a circuit or decomposition. I agree the longer name here makes the PEP violation more visible, but at the same time `givens_rotation` would stand out among other gates even more. Compare:

```python
yield Rx(np.pi/2).on(a)
yield Ry(np.pi/2).on(b)
yield GivensRotation(np.pi/8).on(a, b)
yield CNOT(a, b)
```

and

```python
yield Rx(np.pi/2).on(a)
yield Ry(np.pi/2).on(b)
yield givens_rotation(np.pi/8).on(a, b)
yield CNOT(a, b)
```

I don't have a strong preference, but I think the former looks better.

Reply:
Yes, that is the source of the "Hmmm". Leave it as is.